File: java/parser/src/main/java/gov/lbl/parser/parser/AnnotationVisitor.java
package gov.lbl.parser.parser;

import gov.lbl.antlr4.visitor.modelicaBaseVisitor;
import gov.lbl.antlr4.visitor.modelicaParser;
import gov.lbl.parser.domain.Annotation;
import gov.lbl.parser.domain.Class_modification;

public class AnnotationVisitor extends modelicaBaseVisitor<Annotation> {
    @Override
    public Annotation visitAnnotation(modelicaParser.AnnotationContext ctx) {
        Class_modificationVisitor class_modificationVisitor = new Class_modificationVisitor();
        Class_modification class_modification = ctx.class_modification().accept(class_modificationVisitor);
        return new Annotation(class_modification);
    }
}

File: jsParser/parser/AnnotationVisitor.js
const modelicaVisitor = require('../antlrFiles/modelicaVisitor').modelicaVisitor
const Annotation = require('../domain/Annotation')

const Class_modificationVisitor = require('./Class_modificationVisitor')

class AnnotationVisitor {
  constructor () {
    modelicaVisitor.call(this)
    return this
  }

  visitAnnotation (ctx) {
    let class_modification = null

    if (ctx.class_modification()) {
      const class_modificationVisitor = new Class_modificationVisitor.Class_modificationVisitor()
      class_modification = class_modificationVisitor.visitClass_modification(ctx.class_modification())
    }
    return new Annotation.Annotation(class_modification)
  }
};

AnnotationVisitor.prototype = Object.create(modelicaVisitor.prototype)

exports.visitAnnotation = this.visitAnnotation
exports.AnnotationVisitor = AnnotationVisitor
